### PR TITLE
fix: hide doctors with flag instead of modifying dbc

### PR DESCRIPTION
**Pull request by CaiWillis**
TIS21-2838: Change hide doctor functionality to use boolean flag instead of dbc prefix

### CODE REVIEW

File: application/src/main/java/uk/nhs/hee/tis/revalidation/service/DoctorsForDBService.java [hunk 1]
```diff
@@ -218,14 +216,4 @@ private long getCountUnderNotice() {
   private String getConnectionStatus(final String designatedBody) {
     return (designatedBody == null || designatedBody.equals("")) ? "No" : "Yes";
   }
-
-  private String getHiddenDesignatedBodyCode(String dbCode) {
-    if(dbCode != null && !dbCode.startsWith(hiddenPrefix)){
-      return hiddenPrefix + dbCode;
```
Comment (JosephKelly) [inline application/src/main/java/uk/nhs/hee/tis/revalidation/service/DoctorsForDBService.java:224]: Might be the code smell but it looks like there are no uses of `hiddenPrefix` any more.

File: application/src/main/java/uk/nhs/hee/tis/revalidation/entity/DoctorsForDB.java [hunk 2]
```diff
@@ -67,6 +67,7 @@
   private LocalDate lastUpdatedDate;
   private String designatedBodyCode;
   private String admin;
+  private Boolean existsInGmc;
```
Comment (JosephKelly) [inline application/src/main/java/uk/nhs/hee/tis/revalidation/entity/DoctorsForDB.java:70]: If this field should "always" default to true, not just for `DoctorsForDb::convert`, then you can initialise it by adding ` = true;` here.

**Review by JosephKelly — COMMENTED** [review]
Makes sense to me!

**Review by JosephKelly — DISMISSED** [review]

**Review by jayanta2018 — DISMISSED** [review]

**Review by DorisWongNHS — APPROVED** [review]

File: application/src/main/java/uk/nhs/hee/tis/revalidation/repository/DoctorsForDBRepository.java [hunk 3]
```diff
@@ -39,12 +39,14 @@
...
       + "{ 'doctorLastName' : { '$regex' : ?0, '$options' : 'i'}}, { '_id' : { '$regex' : ?0, '$options' : 'i'}}]}, "
-      + "{ 'designatedBodyCode' : { '$in' : ?1 }}, { 'underNotice' : { '$in' : ?2 }}]}")
+      + "{ 'designatedBodyCode' : { '$in' : ?1 }}, { 'underNotice' : { '$in' : ?2 }},"
+      + "{ 'existsInGmc' : 'true' }]}")
   Page<DoctorsForDB> findByUnderNotice(final Pageable pageable, final String searchQuery,
       final List<String> dbcs, final UnderNotice... underNotice);
 
   @Query(value = "{ '$and' : [{ '$or' : [{'doctorFirstName' : { '$regex' : ?0, '$options' : 'i'}}, "
-      + "{ 'doctorLastName' : { '$regex' : ?0, '$options' : 'i'}}, { '_id' : { '$regex' : ?0, '$options' : 'i'}}]}, { 'designatedBodyCode' : { '$in' : ?1 }}, { '_id' : { '$nin' : ?2 }}]}")
+      + "{ 'doctorLastName' : { '$regex' : ?0, '$options' : 'i'}}, { '_id' : { '$regex' : ?0, '$options' : 'i'}}]}, { 'designatedBodyCode' : { '$in' : ?1 }}, { '_id' : { '$nin' : ?2 }},"
```
Comment (JosephKelly) [inline application/src/main/java/uk/nhs/hee/tis/revalidation/repository/DoctorsForDBRepository.java:48]: I'm happy if you don't but you could split this line just before `{ 'designatedBodyCode'...` to be more like the query above.

**Review by JosephKelly — APPROVED** [review]
BSON queries look like they are well-formatted now :sweat_smile:

File: application/src/main/java/uk/nhs/hee/tis/revalidation/repository/DoctorsForDBRepository.java (hunk 3 above)
Comment (CaiWillis) [reply application/src/main/java/uk/nhs/hee/tis/revalidation/repository/DoctorsForDBRepository.java:48]: will do!

**Review by JosephKelly — APPROVED** [review]

**Pull request merged**